src/main/java/ro/unibuc/triplea/application/games/steam/web/SteamGameController.java

```diff
@@ -0,0 +1,44 @@
+package ro.unibuc.triplea.application.games.steam.web;
+
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import ro.unibuc.triplea.application.games.steam.dto.response.SteamGameResponse;
+import ro.unibuc.triplea.domain.games.steam.exception.SteamGameNotFoundException;
+import ro.unibuc.triplea.domain.games.steam.service.SteamGameService;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/v1/games/steam")
+@RequiredArgsConstructor
+@Tag(name = "SteamGames", description = "Steam Game management APIs")
+public class SteamGameController {
+    
+    private final SteamGameService steamGameService;
+
+    @GetMapping("/game-list")
+    public ResponseEntity<List<SteamGameResponse>> getAllGames(@RequestParam(required=false, name="count") Optional<Integer> count) {
+        List<SteamGameResponse> games = steamGameService.getAllGames(count);
+        return ResponseEntity.ok(games);
+    }
+
+    @GetMapping("/{identifier}")
+    public ResponseEntity<?> getGameByIdentifier(@PathVariable String identifier) {
+        Optional<SteamGameResponse> game = steamGameService.getGameByIdentifier(identifier);
+
+        if (game.isPresent()) {
+            return ResponseEntity.ok(game.get());
+        } else {
+            throw new SteamGameNotFoundException("Steam game with identifier " + identifier + " not found");
```

Comment:
Exception should not be treated in controller, move them to service.

Reply:
fixed